File: showly_flutter/lib/features/shows/data/datasources/tvmaze_remote_datasource.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

import '../../../../core/constants/api_config.dart';

class TvMazeRemoteDataSource {
  Future<List<dynamic>> getShows() async {
    final uri = Uri.parse(
      '${ApiConfig.tvMazeBaseUrl}${ApiConfig.showsEndpoint}',
    );

    final response = await http.get(uri);

    if (response.statusCode < 200 || response.statusCode >= 300) {
      throw Exception('Could not fetch TVMaze shows.');
    }

    return jsonDecode(response.body) as List<dynamic>;
  }
}

File: showly-react-native/src/features/shows/data/datasources/tvMazeRemoteDataSource.ts
import { API_CONFIG } from "../../../../core/constants/api";
import { TvMazeShowModel } from "../models/showModel";

export class TvMazeRemoteDataSource {
  async getShows(): Promise<TvMazeShowModel[]> {
    const response = await fetch(
      `${API_CONFIG.TV_MAZE_BASE_URL}${API_CONFIG.SHOWS_ENDPOINT}`
    );

    if (!response.ok) {
      throw new Error("Could not fetch TVMaze shows.");
    }

    return response.json();
  }
}